client/src/components/Rooms.jsx

```
@@ -0,0 +1,150 @@
+'use client';
+
+import { useState } from 'react';
+import { useNavigate } from 'react-router-dom';
+import { useMutation, useQuery, useQueryClient } from '@tanstack/react-query';
+import { MoreVertical } from 'lucide-react';
+
+import {
+  DropdownMenu,
+  DropdownMenuContent,
+  DropdownMenuItem,
+  DropdownMenuTrigger,
+} from '@/components/ui/dropdown-menu';
+
+import './styles.css';
+import { createRoom, deleteRoom, fetchRooms, updateRoom } from '../api/rooms';
+
+export default function Rooms() {
+  const [newRoom, setNewRoom] = useState('');
+  const [editingRoomId, setEditingRoomId] = useState(null);
+  const [editedName, setEditedName] = useState('');
+
+  const navigate = useNavigate();
+  const queryClient = useQueryClient();
+
+  const { data: rooms = [], isLoading } = useQuery({
+    queryKey: ['rooms'],
+    queryFn: fetchRooms,
+  });
+
+  const handleJoinRoom = (roomId) => {
+    navigate(`/rooms/${roomId}`);
+  };
+
+  const createMutation = useMutation({
+    mutationFn: createRoom,
+    onSuccess: () => {
+      queryClient.invalidateQueries(['rooms']);
+      setNewRoom('');
+    },
+  });
+
+  const deleteMutation = useMutation({
+    mutationFn: (id) => deleteRoom(id),
+    onSuccess: () => queryClient.invalidateQueries(['rooms']),
+  });
+
+  const renameMutation = useMutation({
+    mutationFn: ({ id, name }) => updateRoom(id, name),
+    onSuccess: () => {
+      queryClient.invalidateQueries(['rooms']);
+      setEditingRoomId(null);
+      setEditedName('');
+    },
+  });
```

The `renameMutation` also lacks `onError`. Even though you trim the name before calling `mutate`, server-side rename can fail (e.g., duplicate name). Add an `onError` to display errors and avoid silently failing renames.